sensor/kubernetes/localscanner/service_certificates_repository_test.go

```diff
@@ -255,13 +260,43 @@ func (s *serviceCertificatesRepoSecretsImplSuite) TestEnsureServiceCertificateMi
 	s.NoError(err)
 }
 
+func (s *serviceCertificatesRepoSecretsImplSuite) TestEnsureServiceCertificatesForScannerV4() {
+	testutils.MustUpdateFeature(s.T(), features.ScannerV4, true)
+	clientSet := fake.NewSimpleClientset(sensorDeployment)
+	secretsClient := clientSet.CoreV1().Secrets(namespace)
+
+	repo := newServiceCertificatesRepo(sensorOwnerReference()[0], namespace, secretsClient)
+
+	s.NoError(repo.ensureServiceCertificates(context.Background(), scannerV4Certificates))
+}
+
+func (s *serviceCertificatesRepoSecretsImplSuite) TestEnsureCertificatesScannerV4FailureWhenDisabled() {
+	clientSet := fake.NewSimpleClientset(sensorDeployment)
```

Comment:
In case the default value for the feature ever changes, recommend explicitly setting it to `false` for this test.
```suggestion
func (s *serviceCertificatesRepoSecretsImplSuite) TestEnsureCertificatesScannerV4FailureWhenDisabled() {
	testutils.MustUpdateFeature(s.T(), features.ScannerV4, false)
	clientSet := fake.NewSimpleClientset(sensorDeployment)
```